src/game/objects/common.c

```diff
@@ -67,8 +67,9 @@ void Object_DrawPickupItem(ITEM_INFO *item)
     // Save the frame number.
     int16_t old_frame_number = item->frame_number;
     // Modify item to be the anim for inv item and animation 0.
-    item->anim_number = g_Objects[item_num_option].anim_index;
-    item->frame_number = g_Anims[item->anim_number].frame_base;
+    Item_SwitchToAnim(
```

Comment:
I think `Item_SwitchToAnim` will effectively set the anim number to `g_Objects[item->object_number].anim_index + g_Objects[item_num_option].anim_index` in this scenario, and the same with the frame. Is the following needed here instead?

```c
GAME_OBJECT_ID old_object_num = item->object_number;
int16_t old_frame_number = item->frame_number;
item->object_number = item_num_option;
Item_SwitchToAnim(item, 0, 0);
// ...
item->frame_number = old_frame_number;
item->object_number = old_object_num ;
```

Reply:
Oops I think I meant to just write `Item_SwitchToAnim(item, 0, 0);`. It somehow works fine though.

I don't think we need to restore the object number though? `frame_number` is already restored.

Reply:
Well changing it to `Item_SwitchToAnim(item, 0, 0);` without restoring doesn't work. 😆 

![image](https://github.com/LostArtefacts/TR1X/assets/81546780/7bd8f2e5-d1e4-4216-88e8-4da167256b2b)

Reply:
This change doesn't really belong in this commit anyway, so I added it to the to do list above.